libraries/Microsoft.Bot.Builder.LanguageGeneration/TemplatesParser.cs

```diff
@@ -239,115 +191,259 @@ private static LGFileParser.FileContext GetFileContentContext(string text, strin
+            {
+                if (string.IsNullOrWhiteSpace(templateBody))
+                {
+                    var diagnostic = BuildTemplatesDiagnostic(TemplateErrors.NoTemplateBody(templateName), context, DiagnosticSeverity.Warning);
+                    this.templates.Diagnostics.Add(diagnostic);
+                }
+                else
+                {
+                    try
+                    {
+                        return AntlrParseTemplate(templateBody, startLine);
+                    }
+                    catch (TemplateException e)
+                    {
+                        e.Diagnostics.ToList().ForEach(u => this.templates.Diagnostics.Add(u));
+                    }
+                }
+
+                return null;
+            }
+
+            private void CheckTemplateParameters(List<string> parameters, LGFileParser.TemplateNameLineContext context)
+            {
+                foreach (var parameter in parameters)
                 {
-                    var (content, path) = start.ImportResolver(start.Id, id);
-                    if (resourcesFound.All(u => u.Id != path))
+                    if (!IdentifierRegex.IsMatch(parameter))
                     {
-                        Templates childResource;
-                        if (cachedTemplates.ContainsKey(path))
-                        {
-                            childResource = cachedTemplates[path];
-                        }
-                        else
-                        {
-                            childResource = InnerParseText(content, path, start.ImportResolver, start.ExpressionParser, cachedTemplates);
-                            cachedTemplates.Add(path, childResource);
-                        }
-
-                        ResolveImportResources(childResource, resourcesFound, cachedTemplates);
+                        var diagnostic = BuildTemplatesDiagnostic(TemplateErrors.InvalidTemplateName, context);
+                        this.templates.Diagnostics.Add(diagnostic);
                     }
                 }
-                catch (TemplateException err)
+            }
+
+            private void CheckTemplateName(string templateName, ParserRuleContext context)
+            {
+                var functionNameSplitDot = templateName.Split('.');
+                foreach (var id in functionNameSplitDot)
                 {
-                    throw err;
+                    if (!IdentifierRegex.IsMatch(id))
+                    {
+                        var diagnostic = BuildTemplatesDiagnostic(TemplateErrors.InvalidTemplateName, context);
+                        this.templates.Diagnostics.Add(diagnostic);
+                    }
+                }
+            }
+
+            private (string templateName, List<string> parameters) ExtractTemplateNameLine(string templateNameLine)
+            {
+                var hashIndex = templateNameLine.IndexOf('#');
```

Comment:
>templateNameLine.IndexOf('#') [](start = 32, length = 29)

What happens if you have a comment with a # in it?  Does this not get called?

Reply:
just match the "^\s*#.*" pattern